### PR TITLE
chore: improve DenomTrace grpc

### PR DESCRIPTION
**Pull request by catShaark**
## Description

I change the `TraceDenom` gRPC so that it takes in either an `ibc denom` or a `hash` instead of only accept `hash` as input

closes: #101

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -21,37 +21,50 @@ func (suite *KeeperTestSuite) TestQueryDenomTrace() {
...
+			func() {
+				req = &types.QueryDenomTraceRequest{
+					Hash: "!@#!@#!",
+				}
+			},
 			false,
 		},
 		{
-			"success",
+			"false: not found denom trace",
```
Comment (crodriguezvega) [inline modules/apps/transfer/keeper/grpc_query_test.go:59]:
```suggestion
			"failure: not found denom trace",
```

File: modules/apps/transfer/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -21,37 +21,50 @@ func (suite *KeeperTestSuite) TestQueryDenomTrace() {
...
+				suite.chainA.GetSimApp().TransferKeeper.SetDenomTrace(suite.chainA.GetContext(), expTrace)
+
 				req = &types.QueryDenomTraceRequest{
 					Hash: expTrace.Hash().String(),
 				}
 			},
+			true,
+		},
+		{
+			"false: invalid hash",
```
Comment (crodriguezvega) [inline modules/apps/transfer/keeper/grpc_query_test.go:50]:
```suggestion
			"failure: invalid hash",
```

**Review by crodriguezvega — APPROVED** [review]
Nice work, @catShaark! Thank you very much to closing the previous PR and opening this new one.

File: modules/apps/transfer/keeper/grpc_query.go [hunk 3]
```diff
@@ -22,9 +24,16 @@ func (q Keeper) DenomTrace(c context.Context, req *types.QueryDenomTraceRequest)
...
 	}
 
-	hash, err := types.ParseHexHash(req.Hash)
+	var hash tmbytes.HexBytes
+	var err error
+	if strings.HasPrefix(req.Hash, "ibc/") {
+		hash, err = types.ParseHexHash(req.Hash[4:])
+	} else {
+		hash, err = types.ParseHexHash(req.Hash)
+	}
```
Comment (damiannolan) [inline modules/apps/transfer/keeper/grpc_query.go:33]: Have you considered [`strings.TrimPrefix()`](https://pkg.go.dev/strings#TrimPrefix): 

```
hash, err := types.ParseHexHash(strings.TrimPrefix(req.Hash, "ibc/"))
if err != nil {
    // handle
}
```

Comment (catShaark) [reply modules/apps/transfer/keeper/grpc_query.go:33]: ohhh that is much more convenient, thank so much

**Review by damiannolan — APPROVED** [review]
Nice work! Thanks

**Review by colin-axner — APPROVED** [review]
Nice! Could you add a one sentence entry into the [v3-v4 migration docs](https://github.com/cosmos/ibc-go/blob/main/docs/migrations/v3-to-v4.md). It should go under a `## Relayers` section (see migration template for reference)

It should say something like:

```
When using the DenomTrace gRPC, the IBC denomination with the `ibc/` prefix may now be passed in.
```

**Pull request merged**